sdk/core/System.ClientModel/src/Pipeline/ClientPipeline.cs

```diff
@@ -0,0 +1,219 @@
+        //pipelineLength++; // for retry policy
+        //pipelineLength++; // for response buffering policy
+        pipelineLength++; // for transport
+
+        PipelinePolicy[] policies = new PipelinePolicy[pipelineLength];
+
+        int index = 0;
+
+        perCallPolicies.CopyTo(policies.AsSpan(index));
+        index += perCallPolicies.Length;
+
+        if (options.PerCallPolicies != null)
+        {
+            options.PerCallPolicies.CopyTo(policies.AsSpan(index));
+            index += options.PerCallPolicies.Length;
+        }
+
+        int perCallIndex = index;
+
+        // TODO: RetryPolicy will come in a later PR.
+        //if (options.RetryPolicy != null)
+        //{
+        //    policies[index++] = options.RetryPolicy;
+        //}
+        //else
+        //{
+        //    policies[index++] = new RequestRetryPolicy();
+        //}
+
+        perTryPolicies.CopyTo(policies.AsSpan(index));
+        index += perTryPolicies.Length;
+
+        if (options.PerTryPolicies != null)
+        {
+            options.PerTryPolicies.CopyTo(policies.AsSpan(index));
+            index += options.PerTryPolicies.Length;
+        }
+
+        int perTryIndex = index;
+
+        // TODO: Buffering policy will come in a later PR.
+        //TimeSpan networkTimeout = options.NetworkTimeout ?? PipelineResponse.DefaultNetworkTimeout;
+        //ResponseBufferingPolicy bufferingPolicy = new(networkTimeout);
+        //policies[index++] = bufferingPolicy;
+
+        beforeTransportPolicies.CopyTo(policies.AsSpan(index));
+        index += beforeTransportPolicies.Length;
+
+        if (options.BeforeTransportPolicies != null)
+        {
+            options.BeforeTransportPolicies.CopyTo(policies.AsSpan(index));
+            index += options.BeforeTransportPolicies.Length;
+        }
+
+        int beforeTransportIndex = index;
+
+        if (options.Transport != null)
+        {
+            policies[index++] = options.Transport;
+        }
+        else
+        {
+            // TODO: Transport implementation will come in a later PR.
+            //// Add default transport.
+            //policies[index++] = HttpClientPipelineTransport.Shared;
+        }
+
+        return new ClientPipeline(policies, perCallIndex, perTryIndex, beforeTransportIndex); ;
+    }
+
+    public PipelineMessage CreateMessage() => _transport.CreateMessage();
+
+    public void Send(PipelineMessage message)
+    {
+        IReadOnlyList<PipelinePolicy> policies = GetProcessor(message);
+        policies[0].Process(message, policies, 0);
+    }
+
+    public async ValueTask SendAsync(PipelineMessage message)
+    {
+        IReadOnlyList<PipelinePolicy> policies = GetProcessor(message);
+        await policies[0].ProcessAsync(message, policies, 0).ConfigureAwait(false);
+    }
+
+    private IReadOnlyList<PipelinePolicy> GetProcessor(PipelineMessage message)
+    {
+        // TODO: RequestOptions will come in a later PR.
```

Comment:
how are we going to remember to do all the TODOs?

Reply:
These ones will be tracked because ClientModel won't be complete without them!  But, just so we don't lose one: https://github.com/Azure/azure-sdk-for-net/issues/41220